src/main/scala/com/atomist/source/git/Retry.scala

```diff
@@ -0,0 +1,41 @@
+package com.atomist.source.git
+
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.annotation.tailrec
+import scala.util.{Failure, Random, Success, Try}
+
+/**
+  * Generic retry with exponential backoff and jitter.
+  */
+object Retry extends LazyLogging {
+
+  private val Rnd = Random
+
+  /**
+    * Implement retry with exponential backoff and jiggle.  Each subsequent invocation
+    * will wait double + jiggle of the previous.  Defaults for retries and wait result
+    * in a mean retry period of about 50 seconds.
+    *
+    * @param opName name of operation, used for logging
+    * @param n how many attempts to make
+    * @param wait amount of time to wait
+    * @param fn operation to retry
+    * @tparam T return type pf fn
+    * @return return value from successful call of fn
+    */
+  @tailrec
+  private[git] final def retry[T](opName: String, n: Int = 9, wait: Long = 0L)(fn: => T): T = {
```

Comment:
This stuff is notoriously tricky to get right. Looks ok on the face of it, but I don't see any tests - could have missed that though?

Is there not a third-party library we could use for this?